src/output-json-dns.c

```diff
@@ -730,18 +995,34 @@ static void JsonDnsLogInitFilters(LogDnsFileCtx *dnslog_ctx, ConfNode *conf)
     if (conf) {
         const char *query = ConfNodeLookupChildValue(conf, "query");
         if (query != NULL) {
-            if (ConfValIsTrue(query)) {
-                dnslog_ctx->flags |= LOG_QUERIES;
+            if (strcasecmp(query, "unified") == 0) {
+                    dnslog_ctx->flags &= ~LOG_SPLIT;
+                    dnslog_ctx->flags |= (LOG_QUERIES|LOG_UNIFIED);
+            } else if (strcasecmp(query, "split") == 0) {
+                    dnslog_ctx->flags &= ~LOG_UNIFIED;
+                    dnslog_ctx->flags |= (LOG_QUERIES|LOG_SPLIT);
             } else {
-                dnslog_ctx->flags &= ~LOG_QUERIES;
+                if (ConfValIsTrue(query)) {
```

Comment:
With "yes" and "yes", I'm not getting classic style, it looks like I'm getting split and unified. Should the flags be updated here?